TelemetryManager/Inc/TMCircularBuffer.hpp

```diff
@@ -76,6 +76,9 @@ class TMCircularBuffer : public CircularBuffer {
      * @return int The index of the current byte in the queue.
      */
     int currentIndex();
+
+    private: 
+    uint8_t index_ = 0;
```

Comment:
What happens if the read method is used from the base class? The index_ in TMCircularBuffer would not track that right? The RFD900 DMA uses the base class methods, but we access the same buffer via creating a TMCircularBuffer with the common CircularBuffer object. Maybe you could have a getIndex() method in the common CircularBuffer and just have the TMCircularBuffer currentIndex function return that?